spec/models/manageiq/providers/vmware/infra_manager/refresher_storage_profile_spec.rb

```diff
@@ -0,0 +1,111 @@
+require Rails.root.join('spec/tools/vim_data/vim_data_test_helper')
+
+describe ManageIQ::Providers::Vmware::InfraManager::Refresher do
+  before(:each) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems = FactoryGirl.create(
+      :ems_vmware_with_authentication,
+      :zone => zone, :name => "VC41Test-Prod",
+      :hostname => "VC41Test-Prod.MIQTEST.LOCAL",
+      :ipaddress => "192.168.252.14"
+    )
+
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager)
+      .to receive(:connect).and_return(FakeMiqVimHandle.new)
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager)
+      .to receive(:disconnect).and_return(true)
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager)
+      .to receive(:has_credentials?).and_return(true)
+
+    EmsRefresh.refresh(@ems)
+  end
+
+  context "Handling VMware Storage Profile" do
+    context "association with storages" do
+      let(:storage_profile_ref) { 'aa6d5a82-1c88-45da-85d3-3d74b91a5bad' }
+      let(:storage_location) { '4d3f9f09-38b9b7dc-365d-0010187f00da' }
+
+      it 'handles storage profile deletion' do
+        storage_profile     = StorageProfile.find_by(:ems_ref => storage_profile_ref)
```

Comment:
This is present in every spec and they can all be removed to a single helper around line 4:

```
let(:storage_profile) { StorageProfile.find_by(:ems_ref => storage_profile_ref) }
```

Note you can leave line 84 alone (since that's a second check to make sure it's updated properly)